### ISSUE TITLE
I Want to Change NbChatModule component to feet my need

### ISSUE THREAD

**Issue by TnCoders** [akveo/nebular#1596]
Hello,
As I didn't find any way to ask questions, no forum, I'am looking to change the Nebular Chat Ui component to fit My Need:I would like to use Chat Ui to create a simple Chatbot without IA.

For example:
- in the Chat ui the Sender avatar is not Displayed only reply message type that display avatar
- I would like also to put a waiting GIF when both make some APi call, for example and  the sender waits for response.
- I need for example to add the possibilities to have an input button that the boat will suggest to users inside the chat message.

My question how can I Change the Chat Ui component and then install it  with new features ?

**Comment by yggg** [comment]
You need to extend `NbChatMessageComponent`  to customize chat messages.
There is `ngSwitch`  ([source](https://github.com/akveo/nebular/blob/3d2e4dfa9d7f061271abb313fcd5c64a48d366a7/src/framework/theme/components/chat/chat-message.component.ts#L65)) which used to determine what component to render. You can add new message types to add a bot response or anything else you need.

To enable avatars in replies remove `ngIf` [here](https://github.com/akveo/nebular/blob/3d2e4dfa9d7f061271abb313fcd5c64a48d366a7/src/framework/theme/components/chat/chat-message.component.ts#L59).

Also, you need to extend `NbChatComponent` to change selector used in `ng-content` to your custom message selector.

To sum up:
1. Add custom messages components, like bot suggestions.
2. Extend `NbChatMessageComponent` to add support for custom messages created above.
3. Update message selector in `NbChatComponent` template.
4. Create Module where you declare and export your custom components along with reexporting NbChatModule.

**Comment by Miftahunajat** [comment]
@vodz @yggg 
How do i create class that override the component ?
Because I need the component to be included in NbChatModule

**Comment by vodz** [comment]
> I'm looking to do something similar here. I extended `NbChatComponent ` and `NbChatMessageComponent `however the custom child components I created lost all the original nebular theme styles. I tried to link the scss files directly from `node_modules `to my component `styleUrls `but that didn't work.
> 
> How to use the original nebular theme styles in my extended `CustomChatComponent `and `CustomChatMessageComponent`?

I partly solved my problem: if I copy all the styles from `\node_modules\@nebular\theme\styles\core\_mixins.scss` into global `styles.scss` the styles finally work in my extended component. However if I have them in my component stylesheet it seems the `nb-theme()` modifiers don't get recognized and do nothing, only a couple of "native" styles get applied.

Any idea how to fix this please?

**Comment by Miftahunajat** [comment]
Finally I'm extending my NbChatComponent and NbChatMessageComponent by copying the whole `chat` folder to `theme/components/chat` in my directory i also copied the module the whole `NbChatModule` therefore i still have syle from nebular.

**Comment by allanalves** [comment]
I solved the styles issue by extending the three elements:
`
app-chat {
  @extend nb-chat;
}

app-chat-message {
  @extend nb-chat-message;
}

app-chat-message-link-card {
  @extend nb-chat-message-text;
}
`

The chat component ('nb-chat' or 'app-chat' in the case) was not recognizing the size attribute, that I've set to "medium". To solve that, I copied the native code to the 'chat.component.scss' file:

`
:host {
  display: flex;
  flex-direction: column;
  position: relative;
  height: 100%;
}
`

**Comment by FilipK23** [comment]
@allanalves , Please what was the import of @extend nb-chat; ?

**Comment by allanalves** [comment]
> @allanalves , Please what was the import of @extend nb-chat; ?

@FilipK23, I imported these:

@import 'themes';
@import '~@nebular/theme/styles/globals';

Remembering this code is in styles.scss in the src folder.

**Comment by gllrm-andres** [comment]
@allanalves  gracias bro, me funcionó para customizar un componente con los estilos de nebular ❤

**Comment by allanalves** [comment]
> @allanalves gracias bro, me funcionó para customizar un componente con los estilos de nebular ❤

Great! Happy to help.

**Comment by yabad94** [comment]
I'm facing the same problem to customize and add a new nb-chat-message component. I follow @yggg and @allanalves recomendations trying to customize nb-chat-message component in file case but didn't work as expected. 

1. I created an inherent component from NbChatComponent called NbChatCustomComponent.
2. I copied the NbChatComponent's html in the NbChatCustomComponent's html. So nbChatCustomComponent.html contains the same html code than NbChatComponent.
3. I added: `nb-custom-chat { @extend nb-chat; }` to styles.scss
It works at this level but the styles in the inherent component fails (there is no nb-card-body style inside the chat and no vertical scrolling working inside the chat). 
What I am doing wrong?

**Comment by zarghambetterfront** [comment]
Thanks everyone, i was able to extend and create a custom chat component, I am unable to disable animations. any suggestions regarding this ?

**Comment by JustDoItSascha** [comment]
How could you extend the ChatComponent in Nebular 8. When I try to do it, it's complaining about not finding the NbCustomMessageService, but I can't add it to the providers list, because it's not in the public_api.ts and therefore I get `Module not found: Error: Can't resolve '@nebular/theme/components/chat/custom-message.service'` when trying to add it.

**Comment by derekfreed** [comment]
Is there any example code of how to extend the NbChatMessageComponent and NbChatComponent?

**Comment by GiKyouGetsu** [comment]
> 

Same request!
